src/cheri_pte.sail

```diff
@@ -39,7 +85,62 @@ function isInvalidPTE(p : pteAttribs, ext : extPte) -> bool = {
 
 union PTE_Check = {
   PTE_Check_Success : ext_ptw,
-  PTE_Check_Failure : ext_ptw
+  PTE_Check_Failure : (ext_ptw, ext_ptw_fail)
+}
+
+/*
+ * Assuming we're allowed to load from this page, modulate our cap response
+ */
+val checkPTEPermission_LC : (PTE_Bits, Ext_PTE_Bits, ext_ptw) -> PTE_Check effect { escape, rreg }
+function checkPTEPermission_LC(p, e, ext_ptw) = {
+  let lcav : ext_ptw_lc =
+    if e.CapRead() == 0b0
+    then match (e.CapRead_Gen(), e.CapRead_Mod()) {
+           (0b1, _)  => not_implemented("Reserved PTE extension bits: !LC but Gen"),
```

Comment:
Does not_implemented imply a simulator crash? Surely we want to define some behaviour here, e.g. trap always?